loopchain/blockchain/blockchain.py

```diff
@@ -1101,11 +1117,19 @@ def genesis_invoke(self, block: Block, prev_block_ = None) -> ('Block', dict):
         self.__invoke_results[new_block.header.hash] = (tx_receipts, None)
         return new_block, tx_receipts
 
+    def _convert_reps_change_reason(self, state: str) -> NextRepsChangeReason:
+        if state == "0x0":
+            return NextRepsChangeReason.TermEnd
+        elif state == "0x1":
+            return NextRepsChangeReason.Penalty
+
+        return NextRepsChangeReason.NoChange
+
```

Comment:
```suggestion
class NextRepsChangeReason(IntEnum):
    NoChange = -1
    TermEnd = 0
    Penalty = 1

    @classmethod
    def get_member(cls, value: Union[str, int]):
        if isinstance(value, str):
            int_value = int(value, 0)
        elif isinstance(value, int):
            int_value = value
        else:
            # return cls.NoChange or
            raise TypeError

        for member in list(cls._member_map_.values()):
            if member == int_value:
                return member
```
And use like this
```suggestion
block_builder.next_reps_change_reason = NextRepsChangeReason.get_item(next_prep["state"])
```
instead of 
```
block_builder.next_reps_change_reason = self._convert_reps_change_reason(next_prep["state"])
```

Reply:
updated!